torch/linalg/__init__.py

```diff
@@ -140,3 +140,49 @@
     >>> LA.norm(m[0, :, :]), LA.norm(m[1, :, :])
     (tensor(3.7417), tensor(11.2250))
 """)
+
+tensorsolve = _add_docstr(_linalg.linalg_tensorsolve, r"""
+linalg.tensorsolve(input, other, dims=None, *, out=None) -> Tensor
+
+Computes a tensor ``x`` such that ``tensordot(input, x, dims=x.ndim) = other``.
+The resulting tensor ``x`` has the shape equal to ``input[other.ndim:]``.
+Before computing ``x``, dimensions of :attr:`input` can be moved using :attr:`dims` to match the shape requirements.
+
+Supports real-valued and, only on the CPU, complex-valued inputs.
+
+.. note:: If :attr:`input` does not satisfy the requirement
+          ``prod(input.shape[other.ndim:]) == prod(input.shape[:other.ndim])``
+          after (optionally) moving the dimensions using using :attr:`dims`, then a RuntimeError will be thrown.
+
+Args:
+    input (Tensor): left-hand-side tensor, it must satisfy the requirement
+                    ``prod(input.shape[other.ndim:]) == prod(input.shape[:other.ndim])``.
+    other (Tensor): right-hand-side tensor of shape ``input.shape[other.ndim]``.
+    dims (Tuple[int]): dimensions of :attr:`input` to be moved to the end keeping the order,
```

Comment:
"to be moved to the end keeping the order, ... " -> "to be moved before the computation. Equivalent to calling \`\`input = movedim(input, dims, range(len(dims) - input.ndim, 0))\`\`. If None (default), no dimensions are moved."